Add OPL tutorial (#423)
oasisprotocol/docs: docs/dapp/opl/introduction.md

---
description: How to build your first dApp on OPL
---

# Overview

> In this tutorial, you will create a [secret ballot](https://en.wikipedia.org/wiki/Secret_ballot)

Reviewer: Since this is the first DAO in our docs, I'd expand the introduction a bit.

```suggestion
On-chain voting is the basis for any decentralized autonomous organization
(DAO) that would like to foster bottom-up decision making.
In this tutorial, you will create a [secret ballot](https://en.wikipedia.org/wiki/Secret_ballot)
```

> dApp that can only be built with the Oasis Privacy Layer.

Reviewer: ```suggestion
dApp that can only be built with the Oasis privacy layer.
```

Reviewer: I was taught that the "Oasis Privacy Layer" is the name of our product and should be treated as a proper noun :shrug: So it's fine to write it as "Oasis Privacy Layer". You can dismiss this and related comments.

Author: 😬

> Why is this important? Privacy protects a DAO token holder from intimidation

Reviewer: ```suggestion
Why is this important? Privacy protects the voter (DAO token holder) from intimidation
```

and bullying when exercising their right of participation on a protocol.
Vote organizers can encourage participation with ballots not only by
protecting the identity of voters but also by sealing the results of an ongoing
vote, giving the same weight to the first and last votes.

## Getting Started

If you have made a dApp before, then you likely know most if not all of
the tools covered here! But even if you haven't used all of these tools listed,
you should still keep going! We will do this together.

By the end of this tutorial, we will have:

- used [Hardhat](https://hardhat.org/docs) to write the smart contracts for an
OPL dApp
- used [Hardhat Deploy](https://github.com/wighawag/hardhat-deploy) to deploy
our smart contracts
- used [Pinata](https://docs.pinata.cloud/what-can-i-learn-here/what-is-pinata)
to store simple JSON data. Not everything has to go on a blockchain.
- built a simple [Vue.JS](https://vuejs.org/guide/introduction.html)
app to interact with our dApp through [MetaMask](https://docs.metamask.io/wallet).

> ***Note***. Command line experience will be helpful.

Reviewer: I wouldn't put it this way. Some may get scared, because command line OMG :)

```suggestion
```
